### PR TITLE
Refactor to another library

### PR DESCRIPTION
**Pull request by uaArsen**
#36

### CODE REVIEW

File: src/parse.ts [hunk 1]
```diff
@@ -371,7 +369,9 @@ function parse(xml: string): Promise<Service> {
       const enumTypes = parseEnumTypes(allEnumTypes, schemas);
 
       resolve({ entitySets: entitySets, version: version, complexTypes, singletons, actions, functions, defaultNamespace, entityTypes, enumTypes });
-    });
+    } catch (e) {
```
Comment (jhorbulyk) [inline src/parse.ts]: Why the new `try ... catch` ?

File: src/parse.ts [hunk 2]
```diff
@@ -22,8 +22,8 @@ function getEntityBaseTypes(entityType, entityTypes) {
   const baseTypes = [];
 
   while (entityType) {
-    const baseTypeName = typeNameFromType(entityType['$']['BaseType'])
-    entityType = entityTypes.find(entity => entity['$']['Name'] == baseTypeName);
+    const baseTypeName = typeNameFromType(entityType._attributes.BaseType);
```
Comment (jhorbulyk) [inline src/parse.ts]: A lot of the `'$'` to `_attributes` can be avoided by tweaking the `attributesKey` value on the conversion library.  Why didn't you do that?

Comment (uaArsen) [reply src/parse.ts]: Well, i dont really understand how to do this. Could you please explain? Also will this tweak add any unnecessary overhead?

File: src/parse.ts (hunk 1 above)
Comment (uaArsen) [reply src/parse.ts]: Removed

File: src/parse.ts (hunk 2 above)
Comment (jhorbulyk) [reply src/parse.ts]: I'm saying if you change 
[`const metadata = xml2js.xml2js(xml, { compact: true, trim: true, alwaysArray: true });`](https://github.com/elasticio/odata2openapi/blob/xml-memory-fix/src/parse.ts#L316) 
to 
`const metadata = xml2js.xml2js(xml, { compact: true, trim: true, alwaysArray: true, attributesKey: '$'  });` 
most of the code changes in this file become unnecessary.

Comment (uaArsen) [reply src/parse.ts]: Fixed. Thank you!

File: package.json [hunk 3]
```diff
@@ -25,14 +25,15 @@
...
-  "homepage": "https://github.com/akorchev/odata2openapi#readme",
+  "homepage": "https://github.com/elasticio/odata2openapi#readme",
   "devDependencies": {
     "@types/node": "^7.0.18",
+    "ts-node": "^8.1.0",
     "typescript": "^2.3.2"
   },
   "dependencies": {
-    "xml2js": "^0.4.17"
+    "xml-js": "^1.6.11"
```
Comment (jhorbulyk) [inline package.json]: In general, we avoid `^` in our pacakge dependencies.

**Review by jhorbulyk — CHANGES_REQUESTED** [review]
One very small thing

File: package.json (hunk 3 above)
Comment (uaArsen) [reply package.json]: Fixed

**Review by jhorbulyk — APPROVED** [review]

**Pull request merged**